compliant_social_django/decorators.py

```diff
@@ -0,0 +1,37 @@
+from .audit.clients import AbstractBaseAuditLogger
+
+
+def create_audit_logs(audit_logger: AbstractBaseAuditLogger):
+    """
+    A decorator which injects audit logs into a social_core backend object.
+
+    Attributes:
+        audit_logger    An instantiated instance of an audit logger which inherits from
+                        compliant_social_django.audit.clients.AbstractBaseAuditLogger
+    """
+    def decorator(backend):
```

Comment:
The backend param should also be typed as we're relying on its interface here.

Reply:
Yea, ill type it to the OAuth2 backend